### ISSUE TITLE
Remove deprecated "type" field from the CPU, Memory, Datadog scalers metadata

### ISSUE THREAD

**Issue by amirschw** [kedacore/keda#2844]
### Proposal

In #2309 we add the global `metricType` field to all scalers.

Specific scalers (CPU, Memory, Datadog) already supported different metric types using their metadata `type` field, which we deprecated in favor of the new global field. The deprecated field should be removed in a future release.

**Comment by tomkerkhove** [comment]
Was the deprecation added to our changelog?

**Comment by amirschw** [comment]
It was added to the scalers' docs and we also log a warning when the deprecated field is used

**Comment by tomkerkhove** [comment]
That's OK, let's make sure it is in the changelog as well then.

**Comment by amirschw** [comment]
Added here: #2848

**Comment by JorTurFer** [comment]
When are we going to remove this? maybe we could follow k8s deprecation policies, WDYT?

**Comment by zroubalik** [comment]
We are still `v1alpha1` so I don't think we need to wait till v3.0 and actually I wouldn't like to wait. What about deprecation period of 3 releases?

**Comment by tomkerkhove** [comment]
Fair enough, but there are a lot more of them AFAIK so better build a whole list.

Another approach is introduce version per scaler for this to have more granular control (see https://github.com/kedacore/keda/issues/613)

**Comment by JorTurFer** [comment]
Based on our deprecation policy, we should remove this in v2.9, so I mark it for v2.10

**Comment by tomkerkhove** [comment]
@JorTurFer No we are not allowed to do this: https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md#autoscaling-triggers-and-their-metadata

**Comment by JorTurFer** [comment]
you're right! Thanks for correct me 🙇

**Comment by wozniakjan** [comment]
per https://github.com/kedacore/keda/pull/6229, for CPU and memory, this will actually be 2.18 :)
